### PR TITLE
UCM: external events API

### PR DESCRIPTION
**Pull request by brminich**
- Some events can be marked as external. In this case no memory
    hooks will be installed when handler is registered for these
    events. User is responsible for calling the corresponding UCM
    function (for instace ucp_vm_munmap) when the event occurs.
  - Memory hook test is updated with two more scenarios:
    1) external events and 2) NO_INSTALL memory event flag.
    Updated scenarious included to Jenkins as well.

### CODE REVIEW

File: src/ucm/api/ucm.h [hunk 1]
```diff
@@ -223,6 +223,28 @@ void ucm_unset_event_handler(int events, ucm_event_callback_t cb, void *arg);
 
 
 /**
+ * @brief Add memory events to the external events list.
+ *
+ * @param [in]  events    Bit-mask of events which are supposed to be handled
+ *                        externally.
+ *
+ * @note Setting a handler for external event will not trigger installing of
```
Comment (yosefe) [inline src/ucm/api/ucm.h]: need to make documentation more clear

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2206/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1544/ for details (Mellanox internal link).

File: src/ucm/api/ucm.h (hunk 1 above)
Comment (shamisp) [inline src/ucm/api/ucm.h]: Please better documentation...

**Comment by shamisp** [comment]
I would add some explicit example how it can be used.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2214/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1551/ for details (Mellanox internal link).

**Comment by brminich** [comment]
@yosefe, @shamisp, please take a look. 
The failure is  #856

**Comment by shamisp** [comment]
Looks good to me. @yosefe ?

**Comment by yosefe** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2217/ for details.

**Comment by yosefe** [comment]
:+1:

**Pull request merged**